### ISSUE TITLE
Enable Git LFS

### ISSUE THREAD

**Issue by Noggog** [Starfield-Community-Patch/Starfield-Community-Patch#263]
We will want to enable and make sure we have some Git LFS space.   If we get any assets that go into `packed` and are binary (.dds, .nif, etc) then we'll want to commit these as Git LFS files so they dont clog the repo

**Comment by Scrivener07** [comment]
Here are some notes for LFS. See link here https://git-lfs.com/
We should not wait until after committing a certain to configure it for LFS. We should configure all possible file types for LFS up front so they don't end up baked into the repository history. After running the `git lfs install` command which is done only once per repository, you simply run the track command for each file you wish to participate in large file storage. We can always easily remove a file from LFS and re-add it as a normally tracked git file. But it is not as possible to remove a binary that has already been baked into history. You can also add a file to LFS that has already been normally committed to git, but that normally committed git history will still be baked in up that the point it was added to LFS.

**Data Plugin**
The data plugins already have a text conversion planned for the workflow. Despite this, we should LFS track these files so that they never get baked into the git history when they are inevitably committed.
```
git lfs track "*.esm"
git lfs track "*.esp"
git lfs track "*.esl"
```

Other file types I feel are worth tracking. We should add all potential binary file types that might sneak into a commit.


**Papyrus**
```
git lfs track "*.pex"
```

**User Interface**
```
git lfs track "*.swf"
git lfs track "*.fla"
git lfs track "*.ai"
```

**Audio**
```
git lfs track "*.wav"
git lfs track "*.xwm"
git lfs track "*.fuz"
git lfs track "*.lip"
```

**Asset 3D**
```
git lfs track "*.psd"
git lfs track "*.pdn"
git lfs track "*.nif"
git lfs track "*.max"
git lfs track "*.bgsm"
git lfs track "*.dds"
git lfs track "*.tga"
```


Now that LFS is setup, users can just use git normally without even being aware of LFS.

**Comment by Noggog** [comment]
> We should not wait until after committing a certain to configure it for LFS

Absolutely agree.  It's too late once the content is already in.

I was thinking we'd have to have a discussion about buying GitLFS storage, but it looks like free plan comes with some:
![image](https://github.com/Starfield-Community-Patch/Starfield-Community-Patch/assets/24981326/85c2f531-5264-4674-813f-dc52e9a27496)

**Comment by Noggog** [comment]
Also, I think es(p/m/l), and pex should just be gitignored completely.   We plan on reconstituting those via Spriggit and pex compilers, so there shouldnt be situations where they're committed to the repo at all.

**Comment by Scrivener07** [comment]
As far as funding anything, check with Picky if the Nexus wants to bankroll anything. If not, Im willing to contribute a sizeable lump sum so that we can make the best decisions knowing we have X$ to spend. Direct message me on Discord if you want to follow up with me about it.

Yea, an ignore is far superior. I dont know what I was thinking.

**Comment by Noggog** [comment]
Perhaps we can fund it with the DP the community patch will get, or something silly like that.  Hoping we just slide into the free tier just fine, though

**Comment by Pickysaurus** [comment]
If we get to a point where we need to pay for storage I'm sure I can sort something!

**Comment by Noggog** [comment]
closed with #276
